cloud-platform/app-deploy.md

```diff
@@ -0,0 +1,332 @@
+
+### Deployment
+
+Deployment files are used to specify core information about an application that is being deployed to the Cloud Platform.
+
+See the contents of the `deployment.yml` file below:
+
+```Yaml
+apiVersion: extensions/v1beta1
+kind: Deployment
+metadata:
+  name: cp-demo-app
+spec:
+  replicas: 1
+  template:
+    metadata:
+      labels:
+        app: cp-demo-app
+    spec:
+      containers:
+      - name: cp-demo-app
+        image: 926803513772.dkr.ecr.us-west-1.amazonaws.com/cloud-platform-demo-app:latest
+        ports:
+        - containerPort: 80
+```
+
+If you are using the file as a template for your own application, replace the `cp-demo-app` tags with ones suited to your application.
+
+For the value of the `image:` key, you will see the Repository URI, which has been provided by ECR.
+
+If you are using the file as a template for your own application, replace the value of `image:` with the Repository URI of your application, found in ECR.
+
+### Service
+
+Service files are used to specify port and protocol information for your application and are also used to bundle together the set of pods created by the deployment.
+
+See the contents of the `service.yml` file below:
+
+```Yaml
+kind: Service
+apiVersion: v1
+metadata:
+  name: cp-app-demo-svc
+  labels:
+    app: cp-app-demo-svc
+spec:
+  ports:
+  - port: 80
+    name: http
+    targetPort: 80
+  selector:
+    app: cp-demo-app
+```
+
+If you are using the file as a template for your own application, replace the `cp-demo-app-svc` tags with ones suited to your application.
+
+Also, ensure the `selector:` app tag is the same as specified in the `deployment.yml`.
+
+### Ingress
+
+Ingress files are to use to define external access to the application.
+
+See the contents of the `ingress.yml` file below:
+
+```Yaml
+apiVersion: extensions/v1beta1
+kind: Ingress
+metadata:
+  name: cp-demo-app-ing
+spec:
+  rules:
+  - host: cp-demo-app.integration.dsd.io
+    http:
+      paths:
+      - path: /
+        backend:
+          serviceName: cp-demo-app-svc
+          servicePort: 80
+```
+
+If you are using the file as a template for your own application, replace the `cp-demo-app-ing` tag with one suited to your application.
+
+Also, ensure the `serviceName:` tag is the same as specified in the `service.yml`.
+
+## Configuring secrets
+
+The following is an example of encoding (configuring) aws access-key credentials in your deployment.
+(See https://kubernetes.io/docs/concepts/configuration/secret/#using-secrets-as-environment-variables
+for detailed information regarding providing base64 values in secret objects to Kuberenetes pods)
+
+Create your AWS Credentials access key (making a note of the aws_access_key_id and aws_secret_access_key)
+
+put the aws key secret and id into a file (in this example demo-v9.k8s) as follows ('thekey' referencing the aws_access_key_id
```

Comment:
The names here could easily be confusing, as the key/value pairs within a `Secret` object are commonly referred to as a `key` and `secret` - so this example has a key called `thekey` with a secret, and another key called `thesecret` with another secret. So it'd be better to use the AWS terminology:

```
data:
  aws_access_key_id: QUNDRVNTX0tFWV9JRA==
  aws_secret_access_key: U0VDUkVUX0FDQ0VTU19LRVk=
```